drud/ddev
# Make mariadb root user work, drop tables before import, fixes #852, fixes #853, fixes #810, fixes #670

## The Problem/Issue/Bug:

This combines fixes for a few small issues:
* Use root mariadb user for ddev import-db (https://github.com/drud/ddev/issues/852)
* Make root user work in db container (https://github.com/drud/ddev/issues/853)
* Drop database and recreate before import so that no extra tables or data is in there (https://github.com/drud/ddev/issues/810)

## How this PR Solves The Problem:

* Changes the default user in the db container to 'root' (password=root)
* Makes the root user actually work. It previously worked from the web container but not the db container!
* Drops the database so it can be created clean on import-db.
* Removes the (long obsolete) import.sh file in the db container.

## Manual Testing Instructions:

The temp pushed mariadb-local image is drud/mariadb-local:20180529_fix_maria_root, but since this version of ddev includes that, you don't actually have to know that.

* Create an extra table named 'pet' by using `ddev ssh` and `CREATE TABLE pet (name VARCHAR(20), owner VARCHAR(20));`. Verify with `SHOW TABLES LIKE 'pet';` Then do a `ddev import-db`. Only the tables in your import sql should then be there. Verify with `SHOW TABLES LIKE 'pet';`, which should now not exist.
* `ddev ssh -s db` and `mysql`. You should be in as root user. Try `mysql -uroot -proot`, it should work.
* Use ddev import-db with a complex import that uses triggers, requiring root privileges to import. I will try to come up with a demo import; somebody gave me one but don't know where it is.


## Automated Testing Overview:

No changes were made here.

## Related Issue Link(s):

OPs #852, #853, #810 

## Release/Deployment notes:
